### ISSUE TITLE
Try to make lines follow the same distance progression as cones

### ISSUE THREAD

**Issue by mlenser** [mlenser/kryx-rpg-issues#602]
Or perhaps 1 tier above.

Problems that I'm trying to solve:
- Distances should ideally be as consistent as they can be. Regular distances are 9, 18, 36, and 90 meters. AoEs, lines, and cones inherently can't use those, but we can try to align them on certain levels.
- As lines/walls get longer, the less impact that extra distance provides. A 9-18 meter line is likely sufficient for the vast majorty of cases.

**Comment by itamarcu** [comment]
Currently, cones are in length:  15, 30, 45, 60, 75

While lines are 5 ft wide and their length is:  20, 40, 60, 80, 100

Could there perhaps also be "beams", that are 10 ft wide and scale:  10, 20, 30, 40, 50?

For the numbers themselves, I'm in favor of both 20-40-60 and 30-45-60 (and even 30-60-90), but I think making lines identical to cones would be bad. The extra "precision" enabled by choosing lines is usually not required, as usually characters who attack in large AoEs will choose to hit 2-3 extra monsters even if they have to also hit an ally (at least in my experience), and cones just cover so much more area at 3+ mana.

One problem we've brought up before is that at the lowest and most common level, a 15 ft cone is worse than a 30 ft line, because the line both covers more space and allows for double the range. 

Do lines even have to scale linearly, though?  Other AoEs (cones, spheres) increase in both dimensions, making the area increase quadratically.  Could lines scale the same way?  Something like... 20, 45, 90, 150, 300?

**Comment by mlenser** [comment]
I updated the issue with details on the problems that I'd like to solve.

One thing I'm thinking of now is having lines/walls be long and allow for their distance to be halved so they can be 3 meters wide instead of 1.5 meters wide.

**Comment by itamarcu** [comment]
I didn't consider walls specifically, but for them, the length is much more useful than for lines.  Walls can be bent around someone to form a circle, or bent around certain objects/creatures to divide the battlefield.  Wider/thicker walls are definitely sometimes useful but I think that this change will mostly help the lines and the damaging walls (e.g. wall of light), not the barrier walls (e.g. wall of stone).

**Comment by DalenWBrauner** [comment]
We've discussed before the flavor of taking a Cone spell and "concentrating" it into a Line. I dig that, and think we should stick with it. If we do, then I think it makes sense that a concentrated Line would extend just a bit further than a cone. This is exactly the progression we currently have (Cone lengths are 75% of Line lengths), so my suggestion would be to either not change it at all, or to adjust that 75% scaling factor.

Of course, this flavor doesn't apply to spells that are _just_ lines (e.g. Walls), but I'm not sure if that matters.

**Comment by mlenser** [comment]
> my suggestion would be to either not change it at all

That doesn't solve either of the problems I'm trying to address.

**Comment by DalenWBrauner** [comment]
Fair enough. For Problem 2, I don't think you have too much of a problem here.
> As lines/walls get longer, the less impact that extra distance provides. A 9-18 meter line is likely sufficient for the vast majorty of cases.

While this is true to an extent, I actually think this applies to nearly all spell AoEs. While the difference in size is pretty great for spells cast at 1-3 mana, I don't think most casters are considering casting a spell up to 4th or 5th level just for the sake of distances. Especially considering the daily limitation of 5 mana spells, and especially with how few characters are going to be casting at that level (and some spells can't even be augmented that high), I don't see this being a substantial issue.

We _could_ think of it as Lines/Walls providing more value than necessary at higher levels, and subsequently reduce those distances, but that feels like the wrong approach. Another interpretation is that Lines/Walls' distances at higher levels have a negligible impact, and we should then increase those distances to make those sizes more impactful. Either way, we have a new problem: either we adjust the 4.5 / 9 / 13.5 / 18 / 22.5 scaling at all mana levels to keep the scaling consistent, and risk throwing off the balance of manacosts 1-3 (which I think are fine-tuned quite nicely), or we only adjust the 4 and 5 mana distances and not worry as much about consistent scaling.

tl;dr: I simply don't think the current scaling can be improved upon to solve that 2nd problem. I think we would need a different approach.

**Comment by DalenWBrauner** [comment]
Good point, I'll dive in more specifically.

> Regular distances are 9, 18, 36, and 90 meters.

Sicne we can't really align AoEs with these distances, AoEs need their own "regular distances".

Right now, excluding Wall Thicknesses, the distribution of distances used for AoE spells are (with M being ManaCost):
- 1.5m = 2 instances: Cylinder Radius at 1M, Sphere Radius at 1M
- 3.0m = 3 instances: Cylinder Radius at 2M, Sphere Radius at 2M, Wall Height at 1M
- **4.5m** = 5 instances
- **6.0m** = 4 instances
- 7.5m = 2 instances: Cylinder Radius at 5M, Sphere Radius at 5M
- **9.0m** = 4 instances
- 12.0 = 2 instances: Line Length at 2M, Wall Height at 4M
- 13.5 = 3 instances: Cone Length at 3M, Cylinder Height at 3M, Wall Length at 3M
- 15.0 = 1 instance: Wall height at 5M
- **18.0** = 4 instances
- 22.5 = 3 instances: Cone Length at 5M, Cylinder Height at 5M, Wall Length at 5M
- 24.0 = 1 instance: Line Length at 4M
- 30.5 = 1 instance: Line Length at 5M

I see the concern here- our only good candidates for "Regular Distances" in the AoE world are 4.5, 6.0, 9.0 and 18.0.

Let's look at your earlier suggestion, which I realize I misread earlier (hence my previous post, now deleted):
> having lines/walls be long and allow for their distance to be halved so they can be 3 meters wide instead of 1.5 meters wide.

Let's suppose we take that and allow for Lines to be twice as long as cones by default, and the same length as cones if we increase the width. If we include both line lengths, that changes our distribution to:

- 1.5m = 2 instances: Cylinder Radius at 1M, Sphere Radius at 1M
- 3.0m = 3 instances: Cylinder Radius at 2M, Sphere Radius at 2M, Wall Height at 1M
- **4.5m** = 6 instances
- 6.0m = 3 instances: Cylinder Radius at 4M, Sphere Radius at 4M, Wall Height at 2M
- 7.5m = 2 instances: Cylinder Radius at 5M, Sphere Radius at 5M
- **9.0m** = 6 instances
- 12.0m = 1 instance: Wall Height at 4M
- **13.5m** = 4 instances
- 15.0m = 1 instance: Wall height at 5M
- **18.0m** = 5 instances
- **22.5m** = 4 instances
- 27.0m = 1 instance: Line Length (Full) at 3M
- 36.0m = 1 instance: Line Length (Full) at 4M
- 45.0m = 1 instance: Line Length (Full) at 5M

Our best candidates now are 4.5, 9.0, 13.5, 18.0, 22.5. This _is_ better, but still leaves some room for cleanup, and obviously the full line lengths at 3 mana and beyond aren't helping. Maybe better distances for lines in that case would be "one tier ahead" of cones; 9 / 13.5 / 18 / 22.5 / 27 instead of 9 / 18 / 27 / 36 / 45.

In both cases, I think the easiest cleanup is probably the Wall Heights. I'm not sure how much they matter to you, but if you want to make those more regular as well, I'd suggest something similar to what you discussed with lines: Consider making Walls taller, and then allowing players to reduce their Height to increase their Thickness. Perhaps their default matches their length, and then their heights can be shrunk to double.

Lastly: if you allow players to shrink their line length to increase the width like you suggested, I believe that solves your 'line/wall lengths are usually too long' problem quite nicely.

I've attached CSVs of both of these if you wanna do anything else with that info.
[Kryx Spell AoE Lengths.zip](https://github.com/mlenser/kryx-rpg-issues/files/5543342/Kryx.Spell.AoE.Lengths.zip)
(They include measurements in feet as well for the metric-impaired like myself.)

**Comment by itamarcu** [comment]
Some of my opinions:
- Cylinder height is really not important.  We should be able to align it to anything we want and basically no one would care.  The height of a cylinder could even remain at a consistent 18m, and it would be fine.
- Wall height is *usually* not important.  It mostly matters for cases when a creature attempts to jump/climb over a wall, or when a wall needs to completely block a huge object (e.g. a ship).  
- Repeating my earlier statement - lines don't have to scale linearly.  If you look at spheres, their radius scaling increases their area by a lot (even ignoring the vertical volume increase).  Cones are the same way.  Look at [this cone comparison](https://i.stack.imgur.com/Et5Px.png) - you can see that even an increase from 45 ft long to 60 ft long is useful.  Each +15 ft length increase makes the total area about +50% the area of the previous one. If we want lines to scale their expected number of targets similarly, I think we have to consider something like (in meters): 9, 15, 27, 45, 81.  It should still work with the "half distance double width" option.

**Comment by mlenser** [comment]
Lines now use the same size as Cones. Cones will generally give you more of an effect while lines can help you avoid unwanted targets
